pkg/apis/core/validation/validation.go

```diff
@@ -4382,8 +4382,14 @@ func validatePodResourceConsistency(spec *core.PodSpec, fldPath *field.Path) fie
 	// calculation without modifying AggregateContainerRequests method.
 	aggrContainerReqs := resourcehelper.AggregateContainerRequests(&v1.Pod{Spec: *v1PodSpec}, resourcehelper.PodResourcesOptions{})
 
+	supportedResourceNames := resourcehelper.SupportedPodLevelResources()
```

Comment:
Nice find! Can you add some unit tests for this?

Reply:
Ok, I'll add some appropriate unit tests and get back to you!